### ISSUE TITLE
Pause Observable

### ISSUE THREAD

**Issue by snowtema** [RxSwiftCommunity/RxSwiftExt#162]
### Name and description
I have a `timer` Observable and I need to pause it with one condition: when it'll resume it'll continue to emit elements from an element, which be paused with **initial delay**.

### Example of use
My code:
```swift
let timerActiveSubject = BehaviorSubject<Bool>(value: true)
let timer = Observable<Int>
    .timer(0, period: 1, scheduler: MainScheduler.instance)
    .pausable(timerActiveSubject)

timer
    .subscribe(onNext: { (sec) in
        debugPrint("Timer tick: \(sec)")
    })
    .disposed(by: self.bag)

...

timerActiveSubject.onNext(false)

...

// When I send `true` element to subject, timer observable resume, 
//   but emitted all leaved elements in moment without delay. 
timerActiveSubject.onNext(true) 
```

**Comment by freak4pc** [comment]
@snowtema Did you look into [https://github.com/RxSwiftCommunity/RxSwiftExt#pausablebuffered](pausableBuffered) ?

**Comment by snowtema** [comment]
@freak4pc Yes, I used both forms but it also didn't work for my case. 
I have a timer observable and when I clicked on the button to pause observable it pausing, but when I clicked one more time on the button for continue, observable emits all skipped elements **in one moment** and continue to emit next elements in the normal case. In detail, I paused observable when it was on 15 sec, waiting 10 seconds and continue and instead of getting 16, 17 elements it emits 16-25 sec **in one moment** and continues 26 sec and etc.

**Comment by freak4pc** [comment]
Yup, that's how pausableBuffered works.

The problem with an operator like you're suggesting is that it "lies" about when events are emitted. It means that the stream isn't only paused but is also actually delaying additional emitted items further down the stream. 

I'm not sure how of a popular use case this would be in order to add an operator here, and I feel like we need more community involvement to see if other people would need this enough. What are your thoughts?

**Comment by bobgodwinx** [comment]
@freak4pc I think we should add it a good idea 💡very useful for a jugging and F1 apps 😁

**Comment by snowtema** [comment]
@freak4pc  So, maybe these operators need to be renamed? :) Because now their names differ with their behaviour and it would mislead.

**Comment by freak4pc** [comment]
@snowtema I don't think so ? The emissions are paused and buffered, and then the buffer is "flushed" after un-paused. I don't think the naming is misleading, you might be adding a different meaning to it which is fine - but not conflicting with the fact the name is correct:)
